magento/magento2
# Admin panel navigation improvement

Hello Magento Team

This is new kind of feature for admin panel. It adds additional links for products in several grids following [Customizable Options] grid.





### Description (*)


Functionality for Magento 2 adds missing links in the admin panel for product edit pages. Code follows an idea from the [Customizable Options] section where is [Import options] product grid. [Import options] product grid contains URLs to product's edit page. Unfortunately, in other grids, there is no way to navigate to product edit pages easily. It adds edit URL links to all grids where navigation is needed to open product edit page in new tab.

![configurations_original](https://user-images.githubusercontent.com/4815371/129722456-3f81dee5-310d-4473-9ddb-9a9f4bfad0cc.png)
![add_related_original](https://user-images.githubusercontent.com/4815371/129722453-d5ad4e3e-c8f8-4098-be12-326048cda795.png)
![related_original](https://user-images.githubusercontent.com/4815371/129722455-311a5670-003f-463d-97cb-eee8b14b8763.png)


### Benefits

The administrator saves precious time while working with a large number of products. While adding options to products each time, there is a need to open grid with additional product items. There is no way to go to their edit pages without opening additional browser tabs and searching in other grids. It's complicated and time-consuming. So it adds edit links to the admin panel grids. After clicking in additional link product edit page is opened in a new tab.

![add_products_to_group_new](https://user-images.githubusercontent.com/4815371/129722612-4792bfce-1142-49a6-9f32-39197df26382.png)
![grouped_new](https://user-images.githubusercontent.com/4815371/129722616-243d4f0b-44d3-48d2-8995-63e863454f8d.png)
![products_to_options_new](https://user-images.githubusercontent.com/4815371/129722619-1c27788c-ca99-4608-bddc-3fa1eeb09d4d.png)
![options_new](https://user-images.githubusercontent.com/4815371/129722621-a0e42b59-3dd1-428b-b046-d5bed8a8465e.png)
![associated_new](https://user-images.githubusercontent.com/4815371/129722623-cb1cc0d4-a737-4779-b647-d6e2b8b9157a.png)
![add_related_new](https://user-images.githubusercontent.com/4815371/129722625-8f95e14f-41c5-4520-abf3-63f311330e4e.png)
![related_new](https://user-images.githubusercontent.com/4815371/129722629-8d02b04c-2a3c-45f1-a41f-7850f1f2aa1b.png)

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Open admin panel and go to Catalog -> Products
2. Edit simple product
3. Go to Related Products, Up-Sells, and Cross-Sells section
4. Click Add Related Products
5. Grid should contain links. After clicking product edit page is opened in new tab.
6. Click Up-Sell Products
7. Grid should contain links. After clicking product edit page is opened in new tab.
8. Click Cross-Sell Products
9. Grid should contain links. After clicking product edit page is opened in new tab.
10. After saving all three grids for Related, Up-Sell and Cross-Sell grids should contain links. After clicking product edit page is opened in new tab.
11. Go back to Catalog -> Products
12. Open bundle product
13. Go to Bundle Items section
14. Open Add Option -> Add Product to Option
15. Grid should contain links. After clicking product edit page is opened in new tab.
16. After saving all options grids should have links. After clicking product edit page is opened in new tab.
17. Go back to Catalog -> Products
18. Open grouped product
19. Go to Grouped Products section
20. Add Products to Group grid should contain links for products. After clicking product edit page is opened in new tab.
21. After saving Grouped Products grid should contain links for products. After clicking product edit page is opened in new tab.
22. Go back to Catalog -> Products
23. Open Configurable product
24. Go to Configurations section
25. Add Products Manually product grid should contain links. After clicking product edit page is opened in new tab.
26. After saving Configurations grid should contain links. After clicking product edit page is opened in new tab.
27. Go to Customizable Options section
28. Import Options grid should contain links. After clicking product edit page is opened in new tab.



### Questions or comments


### Contribution checklist (*)
